pkg/providers/tinkerbell/template.go

```diff
@@ -402,10 +406,23 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, controlPlaneMachineSpec, etcd
 		"externalEtcdVersion":          bundle.KubeDistro.EtcdVersion,
 		"etcdCipherSuites":             crypto.SecureCipherSuitesString(),
 		"kubeletExtraArgs":             kubeletExtraArgs.ToPartialYaml(),
-		"controlPlanetemplateOverride": cpTemplateOverride,
 		"hardwareSelector":             controlPlaneMachineSpec.HardwareSelector,
 		"controlPlaneTaints":           clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints,
 	}
+
+	if clusterSpec.Cluster.Spec.RegistryMirrorConfiguration != nil {
```

Comment:
Perhaps we could move this to its own function such as `populateRegistryMirrorValues(...)`.